script.js

```diff
@@ -1,4 +1,235 @@
+    var result = winningHand(handFromProgram, handFromProgram2);
+  }
+  console.log(userWonCount);
+  console.log(comWonCount);
+
+  var winPercent = (
+    (userWonCount * 100) /
+    (userWonCount + comWonCount)
+  ).toFixed(2);
+
+  //if the first round is a draw, since divide by 0 is NaN
+  if (userWonCount == 0 && comWonCount == 0) {
+    winPercent = 0;
+  }
+  myOutputValue = `Hi ${userName}! You played ${input} while the computer played ${handFromProgram}. <br><br> ${result} <br><br> ${userName}'s winning percentage is ${winPercent}%. <br><br> Let's play again!!!`;
   return myOutputValue;
 };
+
+//to process the game mode
+var gameModeProcess = function (userInput) {
+  if (userInput == 1) {
+    return `normal play`;
+  }
+  if (userInput == 2) {
+    return `reversed play`;
+  }
+  if (userInput == 3) {
+    return `korean SPS`;
+  }
+  if (userInput == 4) {
+    return `computer vs computer`;
+  }
+};
+
+//to validate the user input for the game mode
+var modeValidation = function (userInput) {
+  if (
+    !(
+      userInput == "1" ||
+      userInput == "2" ||
+      userInput == "3" ||
+      userInput == "4"
+    )
+  ) {
+    return true;
+  }
+};
+
+//to validate user input
+var inputValidation = function (userGuess) {
+  if (
+    !(userGuess == "scissors" || userGuess == "stone" || userGuess == "paper")
+  ) {
+    return true;
+  }
+};
+
+//to evaluate the winner for normal play
+var winningHand = function (userInput, programPlay) {
+  if (userInput == programPlay) {
+    return "It is a draw!";
+  }
+  if (
+    (userInput == "scissors" && programPlay == "paper") ||
+    (userInput == "paper" && programPlay == "stone") ||
+    (userInput == "stone" && programPlay == "scissors")
+  ) {
+    userWonCount++;
+    return "Congrats, you win!";
+  }
+  if (
+    (userInput == "stone" && programPlay == "paper") ||
+    (userInput == "scissors" && programPlay == "stone") ||
+    (userInput == "paper" && programPlay == "scissors")
+  ) {
+    comWonCount++;
+    return "Oh no, you lost!";
+  }
+};
+
+//to evaluate the winner for reversed play
+var reversedGameWinningHand = function (userInput, programPlay) {
+  if (
+    (userInput == "scissors" && programPlay == "scissors") ||
+    (userInput == "paper" && programPlay == "paper") ||
+    (userInput == "stone" && programPlay == "stone")
+  ) {
+    return "It is a draw!";
+  }
+  if (
+    (userInput == "scissors" && programPlay == "stone") ||
+    (userInput == "paper" && programPlay == "scissors") ||
+    (userInput == "stone" && programPlay == "paper")
+  ) {
+    userWonCount++;
+    return "You are playing the reversed game. <br><br> Congrats, you win!";
+  }
+  if (
+    (userInput == "stone" && programPlay == "scissors") ||
+    (userInput == "scissors" && programPlay == "paper") ||
+    (userInput == "paper" && programPlay == "stone")
+  ) {
+    comWonCount++;
+    return "You are playing the reversed game. <br><br> Oh no, you lost!";
+  }
+};
```

Comment:
since reversedGameWinningHand using the same logic as winningHand. We don't have to create reversedGameWinningHand. We could change the condition within winningHand()  using if statement.

example
var winningHand = function (userInput, programPlay) {
  var message = "";
  if (userInput == programPlay) {
    message = "It is a draw!";
  }
  if (
    (userInput == "scissors" && programPlay == "paper") ||
    (userInput == "paper" && programPlay == "stone") ||
    (userInput == "stone" && programPlay == "scissors")
  ) {
    userWonCount++;
    message = "Congrats, you win!";
  }
  if (
    (userInput == "stone" && programPlay == "paper") ||
    (userInput == "scissors" && programPlay == "stone") ||
    (userInput == "paper" && programPlay == "scissors")
  ) {
    comWonCount++;
    message = "Oh no, you lost!";
  }
  if (gameMode == "reversed play" && message == "Congrats, you win!") {
    message  = "Oh no, you lost!";
    userWonCount-=1;
    comWonCount++;
    return message;
  }else if(gameMode == "reversed play" && message =="Oh no, you lost!"){
    userWonCount++;
    comWonCount -=1;
    message = "Congrats, you win!"
    return message;
  }
  return message;
}

there are many ways to reuse the same logic for reversed game, changing the parameter is one of them